Telegram backend and frontend

lamkeewei opened the pull request:
## Problem

Support Telegram campaign on backend and frontend. 

## Solution

### Settings Page
**Features**
* User should be able to create a Telegram user credential
* User should be able to delete a Telegram user credential
* User should be able to send a Telegram validation message to a subscribed number
* Associated Telegram bot should support `/updatenumber` command after successfully saving Telegram bot token

Closes #341, closes #350 

### Telegram Campaign Creation
**Features**
* User should be able to select Telegram as campaign type when creating campaign
* User should be able to see created Telegram campaigns on the campaign

Closes #340, closes #349

### Telegram Template Upload
**Features**
* User should be able to upload a message template
* User should be able to upload a message template with line breaks preserved
* User should be able to use [supported HTML tags](https://core.telegram.org/bots/api#html-style) in their message.
* User should not be allowed to use unsupported HTML tags

Closes #342, closes #351 

### Telegram Upload Recipients
**Features**
* User should be able to upload a CSV list of recipients with valid params
* User should not be allowed to upload a CSV list of recipients with missing params
* User should be able to provide country codes in their recipient list
* User should be able to not provide country codes and have the recipient numbers automatically prefixed with the default country code 
* User should be able to preview their templated message with line breaks preserved

Closes #343, closes #352

### Telegram Insert Credentials
**Features**
* User should be able to select a stored user Telegram credentials as the campaign credentials 
* User should be able to provide a new valid Telegram bot token as the campaign credentials
* User should be able to send a validation message to a subscribed recipient after setting the campaign credentials

Closes #344, closes #353

### Telegram Send
**Features**
* User should be able to optionally set their desired send rate for their campaign
* User should be able to trigger a send for their Telegram campaign

Closes #354

### Telegram Campaign Details
**Features**
* User should be able to view campaign statistics
* User should be able to pause the campaign

Closes #346, closes #356

### Miscellaneous
* Update db diagram with Telegram models
* Implement Telegram service classes in frontend and backend 

Closes #338, closes #339,  closes #349

## Deploy Notes
**New environment variables**:
- Backend:
    - `TELEGRAM_BOT_TOKEN` : Test Telegram bot token to use when developing locally
    - `TELEGRAM_WEBHOOK_URL` : Telegram webhook callback URL to configure bots with 
- Frontend:
    - `REACT_APP_GUIDE_SMS_CREDENTIALS_URL` : Renamed guide link for Twilio credentials
    - `REACT_APP_GUIDE_TELEGRAM_CREDENTIALS_URL` : Guide link for Telegram credentials.

**New dependencies**:
- `telegraf` : Library to access Telegram API ([repo](https://github.com/telegraf/telegraf))

## TODOs
- [x] Implement send rate field for Telegram
- [x] Implement temporary mobile friendly design for Settings side navigation
- [x] Update links to guide when #429 is completed
- [x] Return the correct status code when channel type is incorrect
- [x] Fix Telegram mentions in messages
- [ ] Implement download for invalid recipient after #330 is merged
- [ ] Refactor to use shared templating module when it is available

chinying commented:
Largely blackbox tested:
* I think right now the main issues are that the country code related DB entries for recipients don't work with our lambda branch, so I've only been able to test the credentials / recipients by manually inserting DB entries. 
* User should be able to use  [supported HTML tags](https://core.telegram.org/bots/api#html-style)  in their message.
About this point, the inline mention of user doesn’t work, the `href` value gets stripped, although I don't know how likely it is that users will want to include inline mentions in their message templates. 
> 400: Bad Request: can’t parse entities: Expected equal sign in declaration of an attribute of the tag “a” at byte offset 306

* Another thing we should note (at least before merging into staging) will just be any manual DB scripts / queries that have to be run for our schema changes.

lamkeewei commented:
> * For the point on "User should be able to use  [supported HTML tags](https://core.telegram.org/bots/api#html-style)  in their message."
>   About this point, the inline mention of user doesn’t work, the `href` value gets stripped, although I don't know how likely it is that users will want to include inline mentions in their message templates.

I'll take a look at this.

lamkeewei commented:
Fixed the issue with inline mentions for Telegram. There's a [list of accepted url patterns](https://github.com/leizongmin/js-xss/blob/v1.0.6/lib/default.js#L156) in `xss-js` and tg:// is not one of them. 

I've also made a couple of small changes after merging in the updated `telegram` branch:
1. Updated the recipient uploading logic to use the new polling method that #388 introduced
2. Use the new button with Promise support from #430
